lldb/source/Utility/StructuredData.cpp

```diff
@@ -289,3 +292,35 @@ void StructuredData::Null::GetDescription(lldb_private::Stream &s) const {
 void StructuredData::Generic::GetDescription(lldb_private::Stream &s) const {
   s.Printf("%p", m_object);
 }
+
+/// This is the same implementation as `StringRef::split`. Not depending on
+/// `StringRef::split` because it will involve a temporary `SmallVectorImpl`.
+StructuredData::ArraySP StructuredData::Array::SplitString(llvm::StringRef s,
```

Comment:
Although I see the use of this method in the patch, I think it should either land in a different patch or this patch should at least add some unit tests for it.

Reply:
Thanks for the prompt review. I appreciate it.

> or this patch should at least add some unit tests for it.
For sure. Let me add (and also other improvements that you have pointed out).

Reply:
This whole function is a small wrapper around StringRef::split(). I am not sure this specialized method belongs in StructuredData (and I would like to not split the strings for output and error if possible).